tests/integration/test_charm.py

```diff
@@ -426,3 +426,41 @@ async def test_promote_user_admin(
         timeout=5,
     )
     assert res.status_code == 200
+
+
+async def test_anonymize_user(
+    synapse_app: Application,
+    get_unit_ips: typing.Callable[[str], typing.Awaitable[tuple[str, ...]]],
+) -> None:
+    """
+    arrange: build and deploy the Synapse charm, create an user, get the access token and assert
+        that the user is not an admin.
+    act:  run action to promote user to admin.
+    assert: the Synapse application is active and the API request returns as expected.
+    """
+    operator_username = "operator-new"
+    action_register_user: Action = await synapse_app.units[0].run_action(  # type: ignore
```

Comment:
Minor stylistic thing that i do to help type units is to do 'unit: Unit  = next(iter(app.units))'.